ResourceManager.cpp

```diff
@@ -1,42 +1,41 @@
 #include "ResourceManager.h"
 #include "Archives/VolFile.h"
 #include "Archives/ClmFile.h"
-//#include "Streams/StreamReader.h"
+#include "Streams/SeekableStreamReader.h"
 #include "XFile.h"
 
-using namespace std;
 using namespace Archives;
 
-ResourceManager::ResourceManager(const string& archiveDirectory)
+ResourceManager::ResourceManager(const std::string& archiveDirectory)
 {
-	vector<string> volFilenames = XFile::GetFilesFromDirectory(archiveDirectory, ".vol");
+	std::vector<std::string> volFilenames = XFile::GetFilesFromDirectory(archiveDirectory, ".vol");
 
-	for (const string& volFilename : volFilenames) {
-		ArchiveFiles.push_back(make_unique<VolFile>(volFilename.c_str()));
+	for (const auto& volFilename : volFilenames) {
+		ArchiveFiles.push_back(std::make_unique<VolFile>(volFilename.c_str()));
 	}
 
-	vector<string> clmFilenames = XFile::GetFilesFromDirectory(archiveDirectory, ".clm");
+	std::vector<std::string> clmFilenames = XFile::GetFilesFromDirectory(archiveDirectory, ".clm");
 
-	for (const string& clmFilename : clmFilenames) {
-		ArchiveFiles.push_back(make_unique<ClmFile>(clmFilename.c_str()));
+	for (const auto& clmFilename : clmFilenames) {
+		ArchiveFiles.push_back(std::make_unique<ClmFile>(clmFilename.c_str()));
 	}
 }
 
 // First searches for resources loosely in provided directory. 
 // Then, if accessArhives = true, searches the preloaded archives for the resource.
-unique_ptr<SeekableStreamReader> ResourceManager::GetResourceStream(const string& filename, bool accessArchives)
+std::unique_ptr<SeekableStreamReader> ResourceManager::GetResourceStream(const std::string& filename, bool accessArchives)
 {
 	if (XFile::PathExists(filename)) {
-		return make_unique<FileStreamReader>(filename);
+		return std::make_unique<FileStreamReader>(filename);
 	}
 
 	if (!accessArchives) {
 		return nullptr;
 	}
 
-	for (unique_ptr<ArchiveFile>& archiveFile : ArchiveFiles)
+	for (auto& archiveFile : ArchiveFiles)
```

Comment:
We can probably use const here. It would apply to the unique_ptr itself here.

Come to think of it though, the archive objects themselves can probably be marked const, since they are not modified after being opened, and they are opened in the constructor. Modifications are only done during re-packing, which is not handled by members of this class.